sdk/containerregistry/container-registry/src/blob/containerRegistryBlobClient.ts

```diff
@@ -0,0 +1,400 @@
+    if (!options.audience) {
+      throw new Error(
+        "ContainerRegistryClientOptions.audience must be set to initialize ContainerRegistryClient."
+      );
+    }
+
+    const defaultScope = `${options.audience}/.default`;
+    const serviceVersion = options.serviceVersion ?? LATEST_API_VERSION;
+    const authClient = new GeneratedClient(endpoint, serviceVersion, internalPipelineOptions);
+    this.client = new GeneratedClient(endpoint, serviceVersion, internalPipelineOptions);
+    this.client.pipeline.addPolicy(
+      bearerTokenAuthenticationPolicy({
+        credential,
+        scopes: [defaultScope],
+        challengeCallbacks: new ChallengeHandler(
+          new ContainerRegistryRefreshTokenCredential(authClient, defaultScope, credential)
+        ),
+      })
+    );
+  }
+
+  /**
+   * Delete a blob.
+   * @param digest - the digest of the blob to delete
+   * @param options - optional configuration used to send requests to the service
+   */
+  public async deleteBlob(digest: string, options?: DeleteBlobOptions): Promise<void> {
+    const { span, updatedOptions } = createSpan("ContainerRegistryBlobClient-deleteBlob", options);
+
+    try {
+      await this.client.containerRegistryBlob.deleteBlob(
+        this.repositoryName,
+        digest,
+        updatedOptions
+      );
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifest - the manifest to upload. If a resettable stream (a factory function that returns a stream) is provided, it may be called multiple times. Each time the function is called, a fresh stream should be returned.
+   */
+  public async uploadManifest(
+    manifest: (() => NodeJS.ReadableStream) | NodeJS.ReadableStream | OciManifest,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult> {
+    const { span, updatedOptions } = createSpan(
+      "ContainerRegistryBlobClient-uploadManifest",
+      options
+    );
+
+    try {
+      let manifestBody: Buffer | NodeJS.ReadableStream;
+
+      if (isReadableStream(manifest)) {
+        manifestBody = await readStreamToEnd(manifest);
+      } else if (typeof manifest === "function") {
+        manifestBody = await readStreamToEnd(manifest());
+      } else {
+        const serialized = serializer.serialize(Mappers.OCIManifest, manifest);
+        manifestBody = Buffer.from(JSON.stringify(serialized));
+      }
+
+      const tagOrDigest = options?.tag ?? (await calculateDigest(manifestBody));
+
+      const { dockerContentDigest } = await this.client.containerRegistry.createManifest(
+        this.repositoryName,
+        tagOrDigest,
+        manifestBody,
+        { contentType: KnownManifestMediaType.OciManifestMediaType, ...updatedOptions }
+      );
+
+      if (!dockerContentDigest) {
+        throw new Error("Digest not provided");
```

Comment:
I'm a little confused about under what conditions this could fail to populate, what would a developer be able to do with this error? Would it be better for us to return whatever the server did (e.g. an empty string?)

Reply:
The header should always be present, but the generated client has it as `string | undefined`, so this check is just to get around that. I can just do a non-null assertion if you think that's better?

Reply:
I think if we mark it as required in swagger and regenerate it should do the right thing? But yeah, maybe an assertion is okay too

Reply:
I was just trying that but unfortunately it appears you can't mark headers in the response as required. Assertion it is, I suppose...